lib/features/children/family_history/models/allowance.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:givt_app/features/children/family_history/models/history_item.dart';
+
+class Allowance extends HistoryItem {
+  const Allowance({
+    required super.amount,
+    required super.date,
+    required super.type,
+    required super.name,
+  });
+
+  @override
+  List<Object?> get props => [amount, date, type];
```

Comment:
shall we add name here as well?

Reply:
why shall we override it at all, since we do not add more fields than parent class has?